### PR TITLE
Send more data to ODT gateway

### PR DESCRIPTION
**Pull request by oxve**
PR number and workflow run url makes debugging easier.
Bonus: add missing timeout fields from the server proto.

b/311260244
b/317441925

### CODE REVIEW

File: tools/on_device_tests_gateway.proto [hunk 1]
```diff
@@ -52,10 +49,14 @@ message OnDeviceTestsCommand {
   string unittest_shard_index = 17;
   string test_attempts = 18;
   string retry_level = 19;
+  string start_timeout = 20;
+  string test_timeout = 21;
+  string builder_url = 22;
```
Comment (oxve) [inline tools/on_device_tests_gateway.proto:54]: Don't merge until corresponding change is deployed on the server side

File: .github/actions/on_device_tests/action.yaml [hunk 2]
```diff
@@ -57,18 +57,21 @@ runs:
...
         GITHUB_COMMIT_AUTHOR_USERNAME: ${{ github.event.commits[0].author.username }}
         GITHUB_COMMIT_AUTHOR_EMAIL: ${{ github.event.commits[0].author.email }}
+        GITHUB_PR_NUMBER: ${{ github.event.pull_request.number }}
+        GITHUB_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
       run: |
         set -uxe
         python3 tools/on_device_tests_gateway_client.py \
           --token ${GITHUB_TOKEN} \
-          --change_id "${GITHUB_PR_HEAD_SHA:-$GITHUB_SHA}" \
+          --change_id ${GITHUB_PR_NUMBER:-postsubmit} \
```
Comment (andrewsavage1) [inline .github/actions/on_device_tests/action.yaml:66]: Is the default just the string "postsubmit"? Why don't we want to have the hash for postsubmit runs?

Comment (oxve) [reply .github/actions/on_device_tests/action.yaml:66]: We do, it's in a label below. Besides, the  PR number mapped better to change_id than the sha does.

File: .github/actions/on_device_tests/action.yaml [hunk 3]
```diff
@@ -57,18 +57,21 @@ runs:
...
           trigger \
           --test_type ${{ env.TEST_TYPE }} \
           --platform ${{ matrix.target_platform }} \
           --config ${{ matrix.config }} \
           --tag cobalt_github_${GITHUB_EVENT_NAME} \
           --builder_name github_${{ matrix.platform }}_tests \
           --build_number ${GITHUB_RUN_NUMBER} \
+          --builder_url GITHUB_WORKFLOW_RUN \
```
Comment (oxve) [inline .github/actions/on_device_tests/action.yaml:74]: Just noticed that I forgot to wrap this in curlies. Will fix.

**Review by andrewsavage1 — APPROVED** [review]

File: .github/actions/on_device_tests/action.yaml (hunk 2 above)
Comment (isarkis) [reply .github/actions/on_device_tests/action.yaml:66]: Are we considering nightlies as postsubmits?

Comment (oxve) [reply .github/actions/on_device_tests/action.yaml:66]: We would, yes, but I don't think it's an important distinction in this case as the field only ever ends up in logs anyway.

File: .github/actions/on_device_tests/action.yaml (hunk 3 above)
Comment (isarkis) [reply .github/actions/on_device_tests/action.yaml:74]: this should be ${GITHUB_RUN_URL} not ${GITHUB_WORKFLOW_RUN}, right?

Comment (oxve) [reply .github/actions/on_device_tests/action.yaml]: Done

**Review by isarkis — APPROVED** [review]

File: tools/on_device_tests_gateway.proto (hunk 1 above)
Comment (kaidokert) [reply tools/on_device_tests_gateway.proto:54]: Has it been deployeth ?

**Pull request merged**